DataFormats/DetId/interface/DetId.h

```diff
@@ -38,7 +38,10 @@ class DetId {
   /// get the detector field from this detid
   constexpr Detector det() const { return Detector((id_>>kDetOffset)&kDetMask); }
   /// get the contents of the subdetector field (not cast into any detector's numbering enum)
-  constexpr int subdetId() const { return ((id_>>kSubdetOffset)&kSubdetMask); }
+  constexpr int subdetId() const { 
+    return ((HGCalEE==det()) || (HGCalHSi==det()) || (HGCalHSc==det()) ?
```

Comment:
@bsunanda this is a top level trick to cope with the fact that the HGCal numbering scheme is in practice distorting the regular use of DetId

Reply:
Yes it is. We have to do that because accessing CaloSubdetectorGeometry from CaloGeometry needs that subdetid for the new HGCal detector should be 0